src/unittest/python/test_barcode_manager.py

```diff
@@ -0,0 +1,115 @@
+"""
+----------------------------------------------------------------------
+Authors: Stephan Nell
+----------------------------------------------------------------------
+Unit tests for the Barcode Manager
+----------------------------------------------------------------------
+"""
+import pytest
+import numpy as np
+import cv2
+import os
+from image_processing.barcode_manager import BarCodeManager
+
+# blank image to test with Height set 1 Width set at 1
+blank_image = np.zeros((1, 1), dtype=np.uint8)
+
+TEMPLATE_DIR = "{base_path}/../../main/python/image_preprocessing/templates/".format(
+    base_path=os.path.abspath(os.path.dirname(__file__)))
+test_image_colour = cv2.imread(TEMPLATE_DIR + "pp2.jpg")
+thanks_obama = cv2.imread(TEMPLATE_DIR + "obama.jpg")
+# Soap Joe official example for Fraud Detection for South African ID books
+soap_joe = cv2.imread(TEMPLATE_DIR + "soapJoeExample.jpg")
+
+cv2 = pytest.importorskip("cv2")
+
+
+def test_barcode_detect():
+    """
+    Tests with wrong parameters
+    """
+    bar_manager = BarCodeManager()
+    with pytest.raises(TypeError):
+        bar_manager.detect(12)
+
+
+def test_barcode_detect_2():
+    """
+    Tests with no parameters
+    """
+    bar_manager = BarCodeManager()
+    with pytest.raises(TypeError):
+        bar_manager.detect()
+
+
+def test_barcode_detect_3():
+    """
+    Tests with correct parameters but with no barcode
+    """
+    cv2.setNumThreads(0)
```

Comment:
There is a problem with the Pybuilder implementation, we need to deactivate OpenCV concurrency functionality, to avoid race conditions. If we don't deactivate the concurrency the testing will simply hang.